keps/sig-scheduling/2779-support-custom-queue-for-the-scheduler/README.md

```diff
@@ -119,52 +124,52 @@ The codes will be updated as following:
 1. User implements a custom queue with the interface SchedulingQueue,
 the name is myQueue, and registers it
 
-  ```
+```
 	command := app.NewSchedulerCommand(
 		app.WithPlugin(coscheduling.Name, coscheduling.New),
-                        …...
+                        ......
 	+	app.WithCustomQueue(myQueue.New),
 	)
-  ```
+```
 
 2. The registered custom queue will be recorded by updated Registry struct.
 
-  ```
+```
   type Registry struct {
     Pf          map[string]PluginFactory
     CustomQueue schedulingQueue.SchedulingQueue
   }
-  ```
+```
 
 3. In function (c *Configurator) create(), pass the custom queue to the scheduler
 if it exists.
-  ```
+```
   if c.registry.CustomQueue != nil {
       podQueue = c.registry.CustomQueue
   } else {
       podQueue = schedulingQueue.NewSchedulingQueue()
   }
-   ```
+```
 
-### Test Plan
-
-<!--
-**Note:** *Not required until targeted at a release.*
+4. Basic inputs for queue's initialization will not be changed, a new interface `Init` is added in
+SchedulingQueue, users must implement it.
 
-Consider the following in developing a test plan for this enhancement:
-- Will there be e2e and integration tests, in addition to unit tests?
-- How will it be tested in isolation vs with other components?
+```
+-func NewPriorityQueue(
++func (p *PriorityQueue) Init(
+ 	lessFn framework.LessFunc,
+ 	informerFactory informers.SharedInformerFactory,
+ 	opts ...Option,
+)
+```
 
-No need to outline all of the test cases, just the general strategy. Anything
-that would count as tricky in the implementation, and anything particularly
-challenging to test, should be called out.
+5. For the scheduler internal packages under ([internal] (https://github.com/kubernetes/kubernetes/tree/b6c75bee15e150628fcc240ab32dba6190d254e4/pkg/scheduler/internal)), queue and heap will be moved out of this folder so that users can access and use the structures inside them, cache and parallelize will not be touched.
```

Comment:
remove the blank between text and URL link.

Again, I don't think it's feasible to move queue and heap out.

Reply:
Item `5` was removed, queue and heap will be internal packages as before.

Reply:
The current blockers in my mind are SchedulingQueue and PriorityQueueOptions (maybe there will be more, I will write some codes to verify it), and I'm planning to move them to pkg\scheduler\framework\interface.go so that the users can access them, this is a right loaction for you?

Reply:
Checked with codes, following items in the current queue need to be exposed, and I'm planning to move them into `pkg\scheduler\framework\interface.go`

```
type PreEnqueueCheck func(pod *v1.Pod) bool

type PriorityQueueOptions struct {}

type Option func(*PriorityQueueOptions)

type SchedulingQueue interface {}
```